phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -108,6 +126,10 @@ public TableResultIterator(MutationState mutationState, Scan scan, ScanMetricsHo
         this.scanIterator = UNINITIALIZED_SCANNER;
         this.renewLeaseThreshold = renewLeaseThreshold;
         this.scanGrouper = scanGrouper;
+        this.hashCacheClient = new HashCacheClient(plan.getContext().getConnection());
+        this.caches = caches;
+        this.retry=plan.getContext().getConnection().getQueryServices().getConfiguration()
```

Comment:
Can we use getProps() here instead of going through the Configuration object? We've found the synchronization done by Configuration to be a bottleneck in the past (and not necessary since these properties don't change).